src/testing/e2e/statefulset.go

```diff
@@ -0,0 +1,334 @@
+package e2e
+
+import (
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+// GenerateStatefulSet is used for testing purposes
```

Comment:
```suggestion
// GenerateStatefulSet prepares an ActiveGate StatefulSet after a Reconciliation of the Dynakube with a specific feature enabled
```
this or something similar would be more descriptive about the purpose of this function